pyomo/repn/ampl.py

```diff
@@ -0,0 +1,1261 @@
+        (False, False): (vars_['less_equal'], vars_['less_equal']),
+    }
+
+
+class text_nl_debug_template(object):
+    unary = {
+        'log': 'o43\t#log\n',
+        'log10': 'o42\t#log10\n',
+        'sin': 'o41\t#sin\n',
+        'cos': 'o46\t#cos\n',
+        'tan': 'o38\t#tan\n',
+        'sinh': 'o40\t#sinh\n',
+        'cosh': 'o45\t#cosh\n',
+        'tanh': 'o37\t#tanh\n',
+        'asin': 'o51\t#asin\n',
+        'acos': 'o53\t#acos\n',
+        'atan': 'o49\t#atan\n',
+        'exp': 'o44\t#exp\n',
+        'sqrt': 'o39\t#sqrt\n',
+        'asinh': 'o50\t#asinh\n',
+        'acosh': 'o52\t#acosh\n',
+        'atanh': 'o47\t#atanh\n',
+        'ceil': 'o14\t#ceil\n',
+        'floor': 'o13\t#floor\n',
+    }
+
+    binary_sum = 'o0\t#+\n'
+    product = 'o2\t#*\n'
+    division = 'o3\t# /\n'
+    pow = 'o5\t#^\n'
+    abs = 'o15\t# abs\n'
+    negation = 'o16\t#-\n'
+    nary_sum = 'o54\t# sumlist\n%d\t# (n)\n'
+    exprif = 'o35\t# if\n'
+    and_expr = 'o21\t# and\n'
+    less_than = 'o22\t# lt\n'
+    less_equal = 'o23\t# le\n'
+    equality = 'o24\t# eq\n'
+    external_fcn = 'f%d %d%s\n'
+    # NOTE: to support scaling and substitutions, we do NOT include the
+    # 'v' or the EOL here:
+    var = '%s'
+    const = 'n%r\n'
+    string = 'h%d:%s\n'
+    monomial = product + const + var.replace('%', '%%')
+    multiplier = product + const
+
+    _create_strict_inequality_map(vars())
+
+
+nl_operators = {
+    0: (2, operator.add),
+    2: (2, operator.mul),
+    3: (2, operator.truediv),
+    5: (2, operator.pow),
+    15: (1, operator.abs),
+    16: (1, operator.neg),
+    54: (None, lambda *x: sum(x)),
+    35: (3, lambda a, b, c: b if a else c),
+    21: (2, operator.and_),
+    22: (2, operator.lt),
+    23: (2, operator.le),
+    24: (2, operator.eq),
+    43: (1, math.log),
+    42: (1, math.log10),
+    41: (1, math.sin),
+    46: (1, math.cos),
+    38: (1, math.tan),
+    40: (1, math.sinh),
+    45: (1, math.cosh),
+    37: (1, math.tanh),
+    51: (1, math.asin),
+    53: (1, math.acos),
+    49: (1, math.atan),
+    44: (1, math.exp),
+    39: (1, math.sqrt),
+    50: (1, math.asinh),
+    52: (1, math.acosh),
+    47: (1, math.atanh),
+    14: (1, math.ceil),
+    13: (1, math.floor),
+}
+
+
+def _strip_template_comments(vars_, base_):
+    vars_['unary'] = {
+        k: v[: v.find('\t#')] + '\n' if v[-1] == '\n' else ''
+        for k, v in base_.unary.items()
+    }
+    for k, v in base_.__dict__.items():
+        if type(v) is str and '\t#' in v:
+            v_lines = v.split('\n')
+            for i, l in enumerate(v_lines):
+                comment_start = l.find('\t#')
+                if comment_start >= 0:
+                    v_lines[i] = l[:comment_start]
+            vars_[k] = '\n'.join(v_lines)
+
+
+# The "standard" text mode template is the debugging template with the
+# comments removed
+class text_nl_template(text_nl_debug_template):
```

Comment:
This class name violates our own standards. I get that this is a legacy thing, but we may want to find places where we violate our own rules at some point.

Reply:
fixed!